### PR TITLE
feat: 24250: Improve Virtual Map detach

### LINKED ISSUE #24250: Improve VirtualMap detach
`VirtualMap.detach() `
- Must be used only when the pipeline is paused, so pausing the pipeline has to be part of this method
- The detach method can be removed from the `VirtualRoot` interface
- Missing some tests

### PR DESCRIPTION
**Pull request by akugal**
`VirtualMap.detach()` 

- Must be used only when the pipeline is paused, so pausing the pipeline has to be part of this method
- The detach method can be removed from the `VirtualRoot` interface
- Missing some tests

Fixes #24250

### CODE REVIEW

File: platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/TeacherPullVirtualTreeView.java [hunk 1]
```diff
@@ -48,20 +46,12 @@ public final class TeacherPullVirtualTreeView extends VirtualTreeViewBase implem
...
      */
-    public TeacherPullVirtualTreeView(
-            final ReconnectConfig reconnectConfig,
-            final VirtualMap map,
-            final VirtualMapMetadata state,
-            final VirtualPipeline pipeline) {
+    public TeacherPullVirtualTreeView(final ReconnectConfig reconnectConfig, final VirtualMap map) {
         // There is no distinction between originalState and reconnectState in this implementation
-        super(map, state, state);
+        super(map, map.getMetadata(), map.getMetadata());
```
Comment (imalygin) [inline platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/TeacherPullVirtualTreeView.java:52]: Is there any reason why we need to pass two identical arguments?

Comment (akugal) [reply platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/TeacherPullVirtualTreeView.java:52]: There is a comment above, and that's how it is passed to this view. 
It's a specific of base view class to hold origin and reconnect metadata, which will probably be refactored with follow-up PRs

**Review by imalygin — APPROVED** [review]
